mqtt 3/1.0.0/modules/ROOT/pages/edits/mqtt3-connector-studio-configuration.adoc

```diff
@@ -0,0 +1,409 @@
+
+[source,xml,linenums]
+----
+<mqtt3:config name="MQTT_FailOver_Config">
+    <mqtt3:fail-over-connection clientId="smart-bentley-123" >
+        <mqtt3:fail-over-servers >
+            <mqtt3:fail-over-url protocol="TCP" host="127.9.0.2" port="1883"/>
+            <mqtt3:fail-over-url protocol="TCP" host="127.0.0.3" port="1884"/>
+            <mqtt3:fail-over-url protocol="TCP" host="127.0.0.1" port="1883"/>
+        </mqtt3:fail-over-servers>
+        </mqtt3:fail-over-connection>
+</mqtt3:config>
+----
+
+[[configure-clean-session]]
+== Configure the Clean Session
+
+In the *MQTT3 Config* window, configure the *Clean session* field to `False` so the broker remembers the client the next time it connects. While the client is offline, all its subscriptions are saved, and Quality of Service (QoS) 1 and 2 messages that the client would want to receive are saved too, until the client reconnects.
+
+Some brokers support the clustering of MQTT brokers in which the nodes share a state. In this case, setting the clean session flag to `False` is useful if the node the connector happens to go offline. This enables the client to reconnect to a different node that is aware of the client's subscriptions so that any messages the connector might have missed while offline are delivered.
+
+If the clean session is set to `True (Default)`, then when the connector disconnects, for whatever reason, all its subscriptions are dropped and the connector has to resubscribe upon reconnection. All messages sent while offline are lost.
+
+In the following example, you configure the clean session:
+
+. In Studio, navigate to the *Global Elements* tab.
+. Click *Create*.
+. In the filter box type `mqtt` and select *MQTT3 Config*.
+. Click *OK*.
+. In the *MQTT3 Config* window, for *Connection*, select *MQTT3 URL Connection*.
+. Set the following fields:
++
+* *Client ID*: `smart-bentley-123`.
+* *URL*: `tcp://127.0.0.1:1883`.
+* *Clean session*: `False`.
+[start=7]
+. Click *OK*.
+
+.MQTT3 Clean Session configuration
+image::mqtt3-cleansession.png[MQTT3 Clean session configuration in Global Element Properties window]
+
+In the XML editor, the `cleanSession` configuration looks like this:
+
+[source,xml,linenums]
+----
+<mqtt3:config name="MQTT_Config">
+  <mqtt3:connection clientId="smart-bentley-123" url="tcp://127.0.0.1:1883">
+      <mqtt3:connection-options cleanSession="false"/>
+    </mqtt3:connection>
+</mqtt3:config>
+----
+
+[[enable-file-persistence]]
+== Enable File Persistence
+
+In the *MQTT3 Config* window, set the *Enable file persistence* field to `True` to enable the MQTT client to persist its state to a file that is used to store any outbound or inbound in-flight messages that the client might have with QoS ≥ 1. In contrast, if you set the field to `False (Default)`, the client state is saved only in memory and in the event of a crash the client is not be able to recover its state.
```

Comment:
If this is a field, doesn't that mean that the user _types_ `True`?
Again, to me anyway, _set_ could be any one of a few actions, like clicking or selecting or providing a value by typing....

Reply:
If we use _QoS ≥ 1_ here, maybe we should earlier as well? I suggest standardizing throughout.
If there are only QoS 1 and 2, this might be a little misleading.

Reply:
```suggestion
In the *MQTT3 Config* window, set the *Enable file persistence* field to `True` to enable the MQTT client to persist its state to a file that is used to store any outbound or inbound in-flight messages that the client might have with QoS ≥ 1. In contrast, if you set the field to `False (Default)`, the client state is saved only in memory and, in the event of a crash, the client cannot recover its state.
```

Reply:
These two sentences ^ are pretty complex, but I could not come up with a graceful way to simplify them.